File: python/githubauthentication/github_authentication_provider.py
from typing import List
from kiota_abstractions.authentication import BaseBearerTokenAuthenticationProvider

from githubauthentication.github_access_token_provider import GitHubAccessTokenProvider

class GitHubAuthenticationProvider(BaseBearerTokenAuthenticationProvider):
	def __init__(
			self,
			client_id: str,
			scope: str,
			allowed_hosts: List[str]= ["api.github.com"]) -> None:
		super().__init__(GitHubAccessTokenProvider(client_id, scope, allowed_hosts))

File: php/src/GitHubAuthentication/GitHubAuthenticationProvider.php
<?php

namespace Baywet\Githubtododemo\Githubauthentication;

use Microsoft\Kiota\Abstractions\Authentication\BaseBearerTokenAuthenticationProvider;

class GitHubAuthenticationProvider extends BaseBearerTokenAuthenticationProvider
{
	public function __construct(string $clientId, string $scope, array $allowedHosts = ["api.github.com"]) {
		parent::__construct(new GitHubAccessTokenProvider($clientId, $scope, $allowedHosts));
	}
}